src/control/cmd/dmg/pool.go

```diff
@@ -370,12 +417,13 @@ func (c *PoolQueryCmd) Execute(args []string) error {
 
 	ctx := context.Background()
 	resp, err := control.PoolQuery(ctx, c.ctlInvoker, req)
-	if err != nil {
-		return errors.Wrap(err, "pool query failed")
-	}
 
 	if c.jsonOutputEnabled() {
```

Comment:
just an idea , but couldn't we add `var wrap string` at the beginning of the `Execute()` methods and `defer func() {if c.jsonOutputEnabled() {return c.errorJSON(err)}; return errors.Wrap(err, wrap)}`, setting wrap var before we return in the body of the `Execute()` method (not setting wrap var will basically make the defer a noop). it might be possible to integrate that behaviour into the json Cmd type to further reduce boilerplate.

Reply:
I considered something like that and a few other approaches, but eventually settled on what is there now because it's explicit. I don't love the extra typing but at least it's easy to follow. I need to get this landed in order to enable the use of dmg in C tests -- we can revisit this in the future if something more clever occurs to us.